### DISCUSSION TITLE
Start the image locally

### DISCUSSION THREAD

**Discussion by fenchi** [ltsp/ltsp#306] [Q&A]
Hello,
I have clients who use LTSP with an image.
The image was created by following this page https://github.com/ltsp/community/wiki/chroots.
But I also have laptops, which I boot from, an image on the hard drive. To do so, I have:
- copied image focal.img and ltsp.img to disk
- Installed grub on the disk
- I configure grub to boot from the image

With Ubuntu Xenial / LTSP5 I had made it work, here is the confugration from grub.
menuentry "LTSP local image" {
   linux   /vmlinuz init=/sbin/init-ltsp root=/dev/sda1 ro loop="xenial.img" quiet splash ip=dhcp
   initrd  /initrd.img
}

Now I try with Ubuntu Focal / LTSP20.06:
menuentry "LTSP local image" {
   linux   /vmlinuz root=/dev/sda1 ro loop="focal.img" quiet splash ip=dhcp
   initrd  /ltsp.img # I tried with and without this line
   initrd  /initrd.img
}

When I boot up I get the following message "failed to start Network Name Resolution", so it boots without a network. And I have GDM which does not launch (I have the simple command line login).

Has anyone ever tried starting without a server and can help me? I am blocked.

**Comment by alkisg** [comment]
Hi, read this: https://ltsp.org/man/ltsp-ipxe/#advanced-image-sources

You don't want loop=, you want ltsp.image=.

Btw I transferred this issue from ltsp/ltsp, which is for bugs,
to ltsp/community, which is for such discussions.

**Comment by fenchi** [comment]
Hello,
Thank you for your answer,

Yes, I have already read https://ltsp.org/man/ltsp-ipxe/#advanced-image-sources
If I replace loop = with ltsp.image, it doesn't get any better. I come across busybox.
But I am using grub and not iPXE because I boot directly from the local hard drive.
How do you go about it? Is it possible to use iPXE instead of GRUB to boot locally without a network?

**Comment by alkisg** [comment]
All ltsp.parameters in /proc/cmdline, like `ltsp.image`, work with any bootloader, like ipxe, grub, syslinux, lilo, whatever.

At that busybox prompt, type: `ls /root`
Do you see /dev/sda1 mounted there?

Then type `ls /root/focal.img`.
Is focal.img there? If not, provide the full path, like `ltsp.image=path/to/focal.img`.

Also type `cat /proc/cmdline` to verify that you have the correct ltsp.image= parameter.

**Comment by fenchi** [comment]
Here is the return of each command line :
(initramfs) `ls /root`
vmlinuz ltsp.img focal.img.zsync
démarrage initrd.img focal.img

(initramfs) `ls /root/focal.img`
/root/focal.img

(initramfs) `cat /proc/cmdline`
BOOT_IMAGE=/vmlinuz root=/dev/sda1 ro ltsp.image=focal.img loop.max_part=9 ip=dhcp

**Comment by alkisg** [comment]
The initramfs-tools package ignores `ip=dhcp` when it sees root=/dev/sd*; it only works with nfsroot, nbdroot etc.

This then will make LTSP unable to configure /etc/network/interfaces, so you're supposed to configure that "manually" and put things like `iface eth0 inet dhcp` there.

**Comment by fenchi** [comment]
I just tried but I still have busybox

**Comment by alkisg** [comment]
Could you upload a video of the boot process, so that we see the error messages?

**Comment by fenchi** [comment]
Voici une vidéo du boot.
https://vimeo.com/449975729

**Comment by alkisg** [comment]
I do not see any LTSP output in this video; are you sure that ltsp.img is loaded?

**Comment by fenchi** [comment]
I am checking.
I am not sure. I redo some tests then I come back to you.

**Comment by fenchi** [comment]
Hello,
Sorry for the absence, I had other projects to advance.

Here I am with my problem.

I partially fixed the problem on startup. The initrd.img overwrites the ltsp.img.
This is a bad grub setup on my part.
You shouldn't write like this:
```
menuentry "LTSP local image" {
 linux /vmlinuz root=/dev/sda1 ro ltsp.image="focal.img" quiet splash
 initrd /ltsp.img
 initrd /initrd.img
}
```

But like this:
```
menuentry "LTSP local image" {
 linux /vmlinuz root=/dev/sda1 ro ltsp.image="focal.img" quiet splash
 initrd /ltsp.img /initrd.img
}
```

The machine starts up correctly, or almost.
There is still the network problem. But I am confident because when I type a "dhclient" the problem is solved.
I will put my solution then I will close this ticket.

**Comment by fenchi** [comment]
I tried to install netplan and configure it, it doesn't work because in the script `ltsp/client/init/45-networking.sh` it removes its configuration.
I also tried with ifupdown (which I had to install) by adding `iface ens160 inet dhcp` in `/etc/network/interfaces` and it does not work either because the script `ltsp/client/init/45-networking.sh` redone then a file `/etc/network/interfaces`.

I have a workaround which is to add the lines below in `/etc/ltsp/ltsp.conf` :
```
[initrd-bottom /]
dhclient
```
But with this solution, if I am not connected and I connect later, it does not require the DHCP request.

Do you know how I should go about it?

**Comment by alkisg** [comment]
The correct way would be for LTSP to be able to function without a server. Currently, 45-networking.sh and other scripts aren't written with that in mind.

You could progress a little with something like:

```shell
[clients]
OMIT_FUNCTIONS="networking_main"
```

...which would invalidate 45-networking.sh, and it would let Ubuntu do automatic network configuration, but I don't know which other scripts could break if networking is unconfigured.

I do not have time to work on making LTSP able to work without a server currently, but patches are welcome.

**Comment by fenchi** [comment]
Great, thank you.
It works and it's clean.
A big thank-you.
